src/backend/core/tasks/metrics.py

```diff
@@ -348,6 +389,94 @@ def map_csv_row(row: Dict[str, str], csv_mapping: Dict[str, str]) -> Dict[str, A
     return mapped_row
 
 
+def store_service_usage_metrics(
```

Comment:
This feels like a lot of code that could be factorized with store_service_metrics ? (same remark for other methods)